src/app/notifications/components/UpdatesAvailable.tsx

```diff
@@ -31,7 +31,7 @@ const UpdatesAvailable = ({className}: {className?: string}) => {
   // if the readyPromise exists, wait for it before rendering
   // if it doesn't exist, we render anyway and in that case
   // onClick() calls activateSwAndReload(undefined) to reload without waiting
-  if (readyPromise && !sw) {
+  if (assertWindow().navigator.serviceWorker && !sw) {
```

Comment:
why did you change this? i mean i think i see how it does the same thing i'm just not sure why you would have to change it

Reply:
Hm yeah not entirely sure what's going on. From the code above, the type of `readyPromise` should be `Promise<whatever> | undefined`.

Reply:
Maybe it's thinking `assertWindow().navigator.serviceWorker` is always defined which is probably not true for older browsers.

Reply:
That seems to be correct; `serviceWorker` is of type `ServiceWorkerContainer`, which is an interface with no `| undefined`. A little odd that it doesn't tell me `serviceWorker` is always defined in the `if`. Taking the test out of the `if` breaks a test (`'doesn\'t render if the serviceWorker or readyPromise change after render()`)

Reply:
Taking the test out would probably break IE and other old stuff: https://caniuse.com/serviceworkers